gapis/perfetto/processor.go

```diff
@@ -35,6 +35,11 @@ type Processor struct {
 func NewProcessor(ctx context.Context, data []byte) (*Processor, error) {
 	p := C.new_processor()
 	log.D(ctx, "[perfetto] Parsing %d bytes", len(data))
+	if len(data) == 0 {
+		log.W(ctx, "[perfetto] Empty profile data.")
+		C.delete_processor(p)
+		return nil, log.Errf(ctx, nil, "profile data is empty.")
+	}
```

Comment:
Put this check above the `C.new_processor()` call, so we can avoid allocating and de-allocating the processor in the error case.